build_defs/python.build_defs

```diff
@@ -322,6 +322,132 @@ def python_test(name:str, srcs:list, data:list|dict=[], resources:list=[], deps:
         tools=[CONFIG.JARCAT_TOOL],
     )
 
+def pip_library(name:str, version:str, hashes:list=None, package_name:str=None,
+                test_only:bool&testonly=False, deps:list=[], post_install_commands:list=None,
+                install_subdirectory:str=None, repo:str=None, use_pypi:bool=None, patch:str|list=None,
+                visibility:list=None, zip_safe:bool=True, licences:list=None, pip_flags:str=None,
+                strip:list=[]):
+    """Provides a build rule for third-party dependencies to be installed by pip.
+
+    Args:
+      name (str): Name of the build rule.
+      version (str): Specific version of the package to install.
+      hashes (list): List of acceptable hashes for this target.
+      package_name (str): Name of the pip package to install. Defaults to the same as 'name'.
+      outs (list): List of output files / directories. Defaults to [name]. If subdirectory is used outs[0] is used as the prefix.
+      test_only (bool): If True, can only be used by test rules or other test_only libraries.
+      deps (list): List of rules this library depends on.
+      post_install_commands (list): Commands run after pip install has completed.
+      install_subdirectory (bool): Forces the package to install into a subdirectory with this name.
+      repo (str): Allows specifying a custom repo to fetch from.
+      use_pypi (bool): If True, will check PyPI as well for packages.
+      patch (str | list): A patch file or files to be applied after install.
+      visibility (list): Visibility declaration for this rule.
+      zip_safe (bool): Flag to indicate whether a pex including this rule will be zip-safe.
+      licences (list): Licences this rule is subject to. Default attempts to detect from package metadata.
+      pip_flags (str): List of additional flags to pass to pip.
+      strip (list): Files to strip after install. Note that these are done at any level.
+    """
+    pip_target_dir = "pip_install"
+
+    package_name = package_name or name
+    package_name = f'{package_name}=={version}'
+    post_install_commands = post_install_commands or []
+    use_pypi = CONFIG.USE_PYPI if use_pypi is None else use_pypi
+    index_flag = '' if use_pypi else '--no-index'
+    pip_flags = pip_flags or CONFIG.PIP_FLAGS
+
+    repo_flag = ''
+    repo = repo or CONFIG.PYTHON_DEFAULT_PIP_REPO
```

Comment:
These config fields need to be migrated to use the plugins version i.e. `CONFIG.PYTHON.*`